source/server/http/admin.h

```diff
@@ -233,7 +258,8 @@ class AdminFilter : public Http::StreamDecoderFilter, Logger::Loggable<Logger::I
   AdminFilter(AdminImpl& parent);
 
   // Http::StreamFilterBase
-  void onDestroy() override {}
+  // TODO (@trabetti) : make more generic (how?)
```

Comment:
What about creating a new public method on `AdminFilter` called `addOnDestroyCallback()`?  This method could take a callback and append to a list of callbacks to execute when the filter is destroyed.  This will only work if we go with what @jmarantz suggested - an architecture where the `AdminFilter` is passed to the handlers directly.

Reply:
Is this the portion that's confusing?  Basically, you'd add a public method to `AdminFilter` that would have the following signature:
```
void addOnDestroyCallback(std::function<void()> cb);
```
That method would append to a list of callbacks that are stored inside the `AdminFilter`.  When `onDestroy()` is called, the `AdminFilter` can call all of the callbacks in that list.  Ultimately, once all this is done, I see your handler looking something like this:
```
Http::Code HystrixSink::handlerHystrixEventStream(... Server::AdminFilter& admin_filter, ....) {
  .
  .
  .
  Http::StreamDecoderFilterCallbacks *stream_decoder_filter_callbacks = admin_filter.getDecoderFilterCallbacks();

  // Separated out just so it's easier to understand
  auto on_destroy_callback = [this, stream_decoder_filter_callbacks] () {
    // Unregister the callbacks from the sink so data is no longer encoded through them.
    unregisterConnection(stream_decoder_filter_callbacks);
  }
  
  // Add the callback to the admin_filter list of callbacks
  admin_filter. addOnDestroyCallback(std::move(on_destroy_callback));
  .
  .
  .
}
```

To do this, you'll need to have the `AdminFilter` be passed to the handlers, and you'll need to add and implement the `addOnDestroyCallback` method.

If you do the above, we can get rid of the `AdminImpl::unregisterHystrixConnection()` and `Server::InstanceImpl:: unregisterHystrixSink`.  This is the last piece of the puzzle to allowing the sink registering itself and requiring no special methods to pass the registration through the server.